festim/temperature/temperature_solver.py

```diff
@@ -75,9 +75,19 @@ def create_functions(self, materials, mesh, dt=None):
         self.v_T = f.TestFunction(V)
 
         if self.transient:
-            ccode_T_ini = sp.printing.ccode(self.initial_value)
-            self.initial_value = f.Expression(ccode_T_ini, degree=2, t=0)
-            self.T_n.assign(f.interpolate(self.initial_value, V))
+            if isinstance(self.initial_condition.value, str):
+                if self.initial_condition.value.endswith(".xdmf"):
+                    with f.XDMFFile(self.initial_condition.value) as file:
+                        file.read_checkpoint(
+                            self.T,
+                            self.initial_condition.label,
+                            self.initial_condition.time_step,
+                        )
+                    self.T_n.assign(self.T)
```

Comment:
I think this can be simplified by simply reading `self.T_n` 

```suggestion
                            self.T_n,
                            self.initial_condition.label,
                            self.initial_condition.time_step,
                        )
```

Reply:
@RemDelaporteMathurin, sure, but it will also require to change new test `test_heat_transfer_create_functions_transient`. I suppose, just removing lines 127-130 would be fine and won't affect the code coverage.